android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/MainActivity.java

```diff
@@ -273,176 +405,268 @@ public void onComplete(Task<Location> task) {
+                AppClient.VerifyLocationReply verifiedLocation =
+                        me.verifyLocation(verifyRequest, dmeHostAddress, port, 10000);
+                Log.i(TAG, "VerifyLocationReply is " + verifiedLocation);
+
+                someText += "[Location Verified: Tower: " + verifiedLocation.getTowerStatus() +
+                        ", GPS LocationStatus: " + verifiedLocation.getGpsLocationStatus() +
+                        ", Location Accuracy: " + verifiedLocation.getGpsLocationAccuracyKm() + " ]\n";
+                List<distributed_match_engine.Appcommon.AppPort> ports = closestCloudlet.getPortsList();
+                String portListStr = "";
+                boolean first = true;
+                String appPortFormat = "{Protocol: %d, Container Port: %d, External Port: %d, Path Prefix: '%s'}";
+                for (Appcommon.AppPort aPort : ports) {
+                    if (!first) {
+                        portListStr += ", ";
 
-                            }
-                            appInstListText += "]]";
-                        }
-                        if (!appInstListText.isEmpty()) {
-                            someText += appInstListText;
-                        }
-                    } else {
-                        someText = "Cannot create request object.\n";
-                        if (!locationVerificationAllowed) {
-                            someText += " Reason: Enhanced location is disabled.\n";
+                    }
+                    portListStr += String.format(Locale.getDefault(), appPortFormat,
+                            aPort.getProto().getNumber(),
+                            aPort.getInternalPort(),
+                            aPort.getPublicPort(),
+                            aPort.getEndPort());
+
+                    //String l7Url = mMatchingEngine.getAppConnectionManager().createUrl(closestCloudlet, aPort, aPort.getPublicPort(), "http", null);
+
+                    String host = aPort.getFqdnPrefix() + closestCloudlet.getFqdn();
+                    int knownPort = 8008;
+                    int serverPort = aPort.getPublicPort() == 0 ? knownPort : aPort.getPublicPort();
+
+                    OkHttpClient client = new OkHttpClient(); //mMatchingEngine.getAppConnectionManager().getHttpClient(10000).get();
+
+                    // Our example server might not like random connections to non-existing /test.
+                    String api = serverPort == knownPort ? "/test" : "";
+                    Response response = null;
+                    try {
+                        Request request = new Request.Builder()
+                                .url("http://" + host + ":" + serverPort + api)
+                                .build();
+                        response = client.newCall(request).execute();
+                        someText += "[Test Server response: " + response.toString() + "]";
```

Comment:
missing \n

sdk test (originally as automation-sdk-porttest)   might be useful here.
<img width="446" alt="Screen Shot 2021-07-01 at 6 56 18 PM" src="https://user-images.githubusercontent.com/24863504/124209386-4a3afd80-da9e-11eb-8a38-0a7ee3afd8cb.png">
https://mobiledgex.atlassian.net/wiki/spaces/SWDEV/pages/1615101953/SDK+Test+App